### PR TITLE
blisp: init at unstable-2023-06-03

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
Tested by flashing a Pinecil v2:
IMG_7084

File: pkgs/development/embedded/blisp/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  cmakeFlags = [
+    "-DBLISP_BUILD_CLI=ON"
+    "-DBLISP_USE_SYSTEM_LIBRARIES=ON"
+  ];
+
+  buildInputs = [ argtable libserialport ]
```
Comment (AndersonTorres) [inline pkgs/development/embedded/blisp/default.nix:21]:
```suggestion
  buildInputs = [ 
    argtable
    libserialport
  ]
```

File: pkgs/development/embedded/blisp/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  src = fetchFromGitHub {
+    owner = "pine64";
+    repo = "blisp";
+    rev = "048a72408218788d519a87bcdfb23bcf9ed91a84";
+    hash = "sha256-hipJrr0D4uEN2hk8ooXeg0gv0X3w4U9ReXbC4oPEPwI=";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  cmakeFlags = [
```
Comment (AndersonTorres) [inline pkgs/development/embedded/blisp/default.nix:16]: Move this after `buildInputs`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/development/embedded/blisp/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "-DBLISP_BUILD_CLI=ON"
+    "-DBLISP_USE_SYSTEM_LIBRARIES=ON"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin $out/lib
+    mv tools/blisp/blisp $out/bin
+    mv shared/libblisp* $out/lib
+    mv static/libblisp* $out/lib
```
Comment (NickCao) [inline pkgs/development/embedded/blisp/default.nix:39]: Aren't these handled by cmake?

Comment (fortuneteller2k) [reply pkgs/development/embedded/blisp/default.nix:39]: Ah yeah, you're right, the stable version didn't for some reason.

**Pull request merged**